File: Number8.java
package algorithm;
import java.util.Scanner;
public class Number8 {

	public static void main(String[] args) {
		// 8. 올바른 괄호
		// 괄호가 입력되면 올바른 괄호이면 'yes' 올바르지 않으면 'no'를 출력합니다.
		
		Scanner stdIn = new Scanner(System.in);
		System.out.print("문자열을 입력하시오:");
		String str = stdIn.nextLine();
		int correct=0;
		for (int i = 0; i < str.length(); i++) {
		    char ch = str.charAt(i);
		    if(ch=='(') correct++;
		    if(ch==')') correct--;
		}
		if(correct==0) System.out.println("YES");
		else System.out.println("NO");
		
	}
}

File: number8.cpp
#include <stdio.h> 

// 올바른 괄호  
/* 괄호가 입력되면 올바른 괄호이면 'YES' 올바르지 않은 괄호이면 'NO'를 출력한다. */ 
 
int main() {
	char a[100];
	int p=0;
	gets(a);
		for(int i=0; a[i]!='\0'; i++){
			if(a[i]=='(')p++;
			if(a[i]==')')p--;	
		}
		if(p==0)printf("YES");
		else printf("NO");
	
	return 0;
}
